tests/functional/test_metamodel/test_model_kwargs.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import unicode_literals
+from textx import (metamodel_from_str)
+import os.path
+from pytest import raises
+from textx.exceptions import TextXError
+
+
+grammar = r"""
+Model: 'MyModel' name=ID;
+"""
+
+text = r"""
+MyModel test1
+"""
+
+
+def test_model_kwargs():
+    mm = metamodel_from_str(grammar)
+    mm._tx_model_kwarg_definitions.add_definition(
+        "parameter1", "an example param (1)"
+    )
+    mm._tx_model_kwarg_definitions.add_definition(
+        "parameter2", "an example param (2)"
+    )
+
+    m = mm.model_from_str(text, parameter1='P1', parameter2='P2')
+    assert m.name == 'test1'
+    assert hasattr(m, '_tx_model_kwargs')
+    assert len(m._tx_model_kwargs) == 2
+    assert len(m._tx_model_kwargs.used_keys) == 0
+
+    assert not m._tx_model_kwargs.have_all_parameters_been_used()
+
+    assert m._tx_model_kwargs['parameter1'] == 'P1'
+    assert len(m._tx_model_kwargs.used_keys) == 1
+    assert 'parameter1' in m._tx_model_kwargs.used_keys
+    assert 'parameter2' not in m._tx_model_kwargs.used_keys
+
+    assert not m._tx_model_kwargs.have_all_parameters_been_used()
+
+    assert m._tx_model_kwargs['parameter2'] == 'P2'
+    assert len(m._tx_model_kwargs.used_keys) == 2
+    assert 'parameter1' in m._tx_model_kwargs.used_keys
+    assert 'parameter2' in m._tx_model_kwargs.used_keys
+
+    assert m._tx_model_kwargs.have_all_parameters_been_used()
+
+    assert m._tx_model_kwargs.get_with_default(
```

Comment:
`get_with_default` should be just `get` where `default` param is optional and by default `None` to make it similar to `dict`

Reply:
That was easy! I learned something very nice.

Reply:
done.